STIPS/STIP-009.md

```diff
@@ -0,0 +1,346 @@
+
+**Reviewer**:
+
+Reviewer: []
+
+## Specification
+
+### ManagerFactory
+
+#### Structs
+
+##### InitializeParams
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the BaseManagerV3|
+|bool|isPending|Bool if manager in pending state|  
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|factory|Address of Set Token factory|
+|mapping(address => InitializeParams)|initialize|Mapping from Set Token to initialization parameters|
+
+
+#### Functions
+
+| Name  | Caller  | Description 	|
+|------	|------	|-------------	|
+|create|deployer|Create new Set Token with a BaseManagerV3 manager|
+|migrate|deployer|Migrate existing Set Token to a BaseManagerV3 manager|
+|initialize|deployer|Initialize modules and extensions|
+
+### BaseManagerV3
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|ISetToken|setToken|Instance of SetToken|
+|mapping(address => bool)|isExtension|Mapping to check if extension is needed|
+|address|owner|Address of owner|
+|address|methodologist|Address of methodologist|
+|address[]|operator|Address of operator(s)|
+
+#### Private Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address[]|extensions|Array of enabled extensions|
+
+#### Functions
+
+| Name  | Caller  | Description 	|
+|------	|------	|-------------	|
+|setManager|owner|Update the manager of the Set Token|
+|addExtension|owner|Add extension to the manager|
+|removeExtension|owner|Remove extension from the manager|
+|interactManager|extension|Interact with a module registered on the Set Token|
+|addModule|owner|Add module to Set Token|
+|removeModule|owner|Remove module from Set Token|
+|setOwner|owner|Update the owner address|
+|setMethodologist|owner|Update the methodologist address|
+|addOperator|owner|Add an operator|
+|removeOperator|owner|Remove an operator|
+
+### BasicIssuanceExtension
+
+#### Structs
+
+##### IssuanceParams
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|IIssuanceModule|issuanceModule|Issuance Module for Set Token|
+|address|manager|Address of the BaseManagerV3|
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|mapping(address=>IssuanceParams)|setIssuanceParams|Mapping from Set Token to issuance parameters|
+
+### StreamingFeeSplitExtension
+
+#### Structs
+
+##### FeeParams
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|IStreamingFeeModule|streamingFeeModule|Streaming Fee Module for Set Token|
```

Comment:
I think in general these modules should be global variables. We don't want to allow managers to somehow use modules we aren't expecting them to